src/AzureClient/Magic/SubmitMagic.cs

```diff
@@ -69,7 +58,14 @@ public override async Task<ExecutionResult> RunAsync(string input, IChannel chan
         {
             var inputParameters = ParseInputParameters(input, firstParameterInferredName: ParameterNameOperationName);
             var operationName = inputParameters.DecodeParameter<string>(ParameterNameOperationName);
-            return await AzureClient.SubmitJobAsync(channel, OperationResolver, operationName);
+
+            var decodedParameters = new Dictionary<string, string>();
+            foreach (var key in inputParameters.Keys)
+            {
+                decodedParameters[key] = inputParameters.DecodeParameter<string>(key);
+            }
```

Comment:
This is great, I always prefer LINQ, thanks for pointing this out! I'm going to apply this change in one of my other PRs since I've already started refactoring this a bit on another branch.